File: boosting/src/learning/prediction/PredictionModel.java
package learning.prediction;

import data.Data;
import data.RegressionData;
import learning.model.ModelWithTeacher;

import java.util.List;

public class PredictionModel implements Predictor {
    private final ModelWithTeacher model;

    public PredictionModel(ModelWithTeacher model) {
        this.model = model;
    }

    @Override
    public void trainAll(List<? extends RegressionData> data) {
        model.trainAll(data);
    }

    @Override
    public void train(RegressionData data) {
        model.train(data);
    }

    @Override
    public double[] output(Data data) {
        return model.output(data);
    }

    public double predictNext(double[] vector) {
        return output(() -> vector)[0];
    }
}

File: boosting/src/learning/prediction/PredictionModel.kt
package learning.prediction

import data.Data
import data.RegressionData
import learning.model.ModelWithTeacher

class PredictionModel(private val model: ModelWithTeacher) : Predictor {

    override fun trainAll(data: List<RegressionData>) {
        model.trainAll(data)
    }

    override fun train(data: RegressionData) {
        model.train(data)
    }

    override fun output(data: Data): DoubleArray {
        return model.output(data)
    }

    override fun predictNext(vector: DoubleArray): Double {
        return output(object : Data {
            override fun asVector(): DoubleArray {
                return vector
            }
        })[0]
    }
}
